trie.h

```diff
@@ -17,9 +17,13 @@
 #   define CDATA(trie, L)  pgm_read_byte(&trie->completions[L])
 #endif
 
-#define TRIE_MATCH_BIT      0x80
-#define TRIE_BRANCH_BIT     0x40
-#define TRIE_CODE_MASK      0x3F
+#define TRIE_MATCH_BIT              0x80
+#define TRIE_BRANCH_BIT             0x40
+#define TRIE_UNCHAINED_MATCH_BIT    0x20
+#define TRIE_EXTENDED_HEADER_BIT    0x10
+#define TRIE_CHAIN_CHECK_COUNT_MASK 0x0F
+#define TRIE_MATCH_SIZE             4
+#define TRIE_CHAINED_MATCH_SIZE     6
```

Comment:
A trie match is represented with 4 bytes:
bytes 1 and 2 contain the bit packing of the `output func`, `backspace count`, and `continuation length`. Bytes 3 and 4 hold the uint16_t index into the continuation array.

A trie chained match is represented with 6 bytes;
The first two bytes represent the uint16_t trie index of the match node of the sub-rule being chained of of. The next 4 bytes are the regular trie match for the chained match and are the same format as specified above. (note, if another rule chains off of *this* rule, it will point to these bytes)